docs/proposals/PROPOSAL_E.md

```diff
@@ -0,0 +1,138 @@
+# Proposal E - Platform Features
+
+This proposal leverages the previously reserved `features` field in the platform to give a finer grain of control to runtimes selecting an image to run.
+
+The goal of this proposal is to define hard requirements that would prevent a binary from being run on a given hardware platform.
+The key differences between proposal A and E are that proposal E:
+
+- Requires each feature to be predefined in the image-spec, rather than reverse DNS namespaced externally managed values.
+- Requires runtimes to be aware of each feature, and reject an index entry with an unrecognized or unsupported feature.
+- Requires runtimes detect support of features directly, rather than from user injected configurations.
+
+## Links
+
+| Description | Link |
+|-------------|------|
+| Image Index Spec | <https://github.com/opencontainers/image-spec/blob/main/image-index.md> |
+
+## Modifications
+
+### Image Spec
+
+The following language is already included in the Image Index definition for the `platform.features`:
+
+> This property is RESERVED for future versions of the specification.
+
+This would replace that text with:
+
+> This OPTIONAL property specifies an array of strings, each specifying a mandatory feature that implementations SHOULD understand.
+> Feature values may be architecture specific.
+
+There would also be a table of known features per architecture, possibly with links to specifications.
+
+An example would also be added to the Image Index:
+
+```jsonc
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 7143,
+      "digest": "sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f",
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 7682,
+      "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux",
+        "features": [ "avx2", "bmi2" ]
```

Comment:
My last question is about this as a list - is it already defined as a list? And the idea is that we'd need an ICD style design of features where every combination of features is represented as a single identifier?

Reply:
Initially it would be a list of CPU features. So you could build a generic amd64/v1 image, and a second image for amd64/v1 plus sse3 to run some application more efficiently.

For context, the OCI platform definition is based on the Go architecture list: https://tip.golang.org/wiki/MinimumRequirements#microarchitecture-support

Reply:
And yes, it is already defined in the spec as a list.